src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h

```diff
@@ -0,0 +1,155 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2017.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey $
+// $Authors: Douglas McCloskey $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_OPENSWATH_MRMFEATUREQC_H
+#define OPENMS_ANALYSIS_OPENSWATH_MRMFEATUREQC_H
+
+#include <OpenMS/KERNEL/MRMFeature.h>
+#include <OpenMS/KERNEL/Feature.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/FORMAT/QcMLFile.h>
+
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+namespace OpenMS
+{
+
+  /**
+
+    @brief The MRMFeatureQC is a class to handle the parameters and options for 
+      MRMFeatureFilter. The format is based on the TraML format.
+
+  */
+  class OPENMS_DLLAPI MRMFeatureQC
+  {
+
+public:
+
+    //@{
+    /// Constructor
+    MRMFeatureQC();
+
+    /// Destructor
+    ~MRMFeatureQC();
+    //@}
+
+    // Members
+
+    /// QCs within a component group
+    struct ComponentGroupQCs
```

Comment:
I think this needs more comments and explanations of what these structs represent and how they relate to one another. If this goes into the public API of OpenMS I think we should add some more information about what a Component is, what a ComponentGroup is etc and how the relate to the TraML entries. This will allow other users to use them correctly.

Reply:
I will go ahead and add more documentation.